internal/graphql/services/common.go

```diff
@@ -0,0 +1,61 @@
+package services
+
+import (
+	"context"
+
+	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
+	"github.com/kubeshop/testkube/pkg/rand"
+)
+
+func Map[T interface{}, U interface{}](list []T, mapper func(item T) U) []U {
+	result := make([]U, len(list))
+	for i, item := range list {
+		result[i] = mapper(item)
+	}
+	return result
+}
+
+func HandleSubscription[T Service, U interface{}](
+	ctx context.Context,
+	topic string,
+	s T,
+	get func() (U, error),
+) (<-chan U, error) {
+	ch := make(chan U, 1)
+
+	// Load initial data
+	initial, err := get()
+	if err == nil {
+		ch <- initial
+	} else {
+		s.Logger().Errorw("failed to get initial data for "+topic, err)
+		return nil, err
```

Comment:
should we close ch there?

Reply:
The channel is not returned, so it will be garbage collected.